src/libbootimg.c

```diff
@@ -171,31 +182,88 @@ void libbootimg_destroy(struct bootimg *b)
     }
 }
 
+uint32_t libbootimg_elf_arrtoval(uint8_t* tab)
+{
+    return (tab[3]<<24) + (tab[2]<<16) + (tab[1]<<8) + (tab[0]);
```

Comment:
This is just little-endian uint32_t, isn't it. Libbootimg doesn't really care about running on big-endian machines, so you can just safely replace the uint8_t[4] arrays with uint32_t in the struct and get rid of this function.